src/utils.c

```diff
@@ -363,3 +363,34 @@ SEXP coerceUtf8IfNeeded(SEXP x) {
   return(ans);
 }
 
+// #4327 this function converts an AsIs object x to list(x) or list(list(x))
+// so that they can be preserved "as is". In addition, the "AsIs" class
+// will be removed.
+SEXP protectAsIs(SEXP x, bool dblist) {
+  if (!inherits(x, "AsIs")) return x; 
+  int nprotect = 0;
+  SEXP oldclass = PROTECT(getAttrib(x, R_ClassSymbol)); ++nprotect;
+  // no need to check the NULL case as inherits(x, "AsIs") == true
```

Comment:
there _may_ technically be an R version thing here. IIRC there was some bugfix in R relatively recently to disallow setting attr on `NULL`. Not sure it's worth the effort to invest in this edge case.

Reply:
The thing is if the `x` is `NULL` then it has already violated the "disallowing setting attr on `NULL`" rule... Looks ok to me to violate it again...

Reply:
I think you're correct